src/nanotron/fp8/meta.py

```diff
@@ -0,0 +1,62 @@
+from dataclasses import dataclass
+from typing import Union
+
+import torch
+import transformer_engine as te  # noqa
+import transformer_engine_extensions as tex
+
+from nanotron.fp8.constants import DTYPE_TO_FP8_MAX
+
+
+@dataclass
+class FP8Meta:
+    """Metadata for FP8Tensor."""
+
+    amax: Union[int, float]
+
+    # TODO(xrsrke): change to Literal[torch.int8, torch.uint8]
+    dtype: torch.dtype
+    inverse_scale: float = 1.0
+
+    @property
+    def te_dtype(self) -> tex.DType:
+        from nanotron.fp8.tensor import convert_torch_dtype_to_te_dtype
+
+        return convert_torch_dtype_to_te_dtype(self.dtype)
+
+    def __post_init__(self):
+        # NOTE: transformer engine only accepts torch tensors
+        # NOTE: remove fill scale factor when initializing
+        # TODO(xrsrke): check if we haven't computed the scaling factor, then do an initial scaling factor
+        # don't set the initial to 1
+        self.scale = 1  # initial scaling factor
+        self.scale = self._compute_scaling_factor()
```

Comment:
Why don't we have `scale` as an attr of `FP8Meta` ?

Reply:
added!